src/Aspire.Hosting.NodeJs/NodeExtensions.cs

```diff
@@ -86,4 +90,132 @@ private static IResourceBuilder<NodeAppResource> WithNodeDefaults(this IResource
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the Vite app.</param>
+    /// <param name="workingDirectory">The working directory of the Vite app.</param>
+    /// <param name="useHttps">When true use HTTPS for the endpoints, otherwise use HTTP.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <example>
+    /// The following example creates a Vite app using npm as the package manager.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// builder.AddViteApp("frontend", "./frontend")
+    ///        .WithNpmPackageManager();
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<ViteAppResource> AddViteApp(this IDistributedApplicationBuilder builder, [ResourceName] string name, string workingDirectory, bool useHttps = false)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+        ArgumentException.ThrowIfNullOrEmpty(workingDirectory);
+
+        workingDirectory = PathNormalizer.NormalizePathForCurrentPlatform(Path.Combine(builder.AppHostDirectory, workingDirectory));
+        var resource = new ViteAppResource(name, "node", workingDirectory);
+
+        var resourceBuilder = builder.AddResource(resource)
+            .WithNodeDefaults()
+            .WithIconName("CodeJsRectangle")
+            .WithArgs(c =>
+            {
+                if (resource.TryGetLastAnnotation<JavaScriptPackageManagerAnnotation>(out var packageManagerAnnotation))
+                {
+                    foreach (var arg in packageManagerAnnotation.RunCommandLineArgs)
+                    {
+                        c.Args.Add(arg);
+                    }
+                }
+                c.Args.Add("dev");
+
+                if (packageManagerAnnotation?.CommandSeparator is string separator)
+                {
+                    c.Args.Add(separator);
+                }
+
+                var targetEndpoint = resource.GetEndpoint("https");
+                if (!targetEndpoint.Exists)
+                {
+                    targetEndpoint = resource.GetEndpoint("http");
+                }
+
+                c.Args.Add("--port");
+                c.Args.Add(targetEndpoint.Property(EndpointProperty.TargetPort));
+            });
+
+        _ = useHttps
+            ? resourceBuilder.WithHttpsEndpoint(env: "PORT")
+            : resourceBuilder.WithHttpEndpoint(env: "PORT");
+
+        return resourceBuilder
+            .PublishAsDockerFile(c =>
+            {
+                c.WithDockerfileBuilder(workingDirectory, dockerfileContext =>
+                {
+                    if (c.Resource.TryGetLastAnnotation<JavaScriptPackageManagerAnnotation>(out var packageManagerAnnotation)
```

Comment:
What happens when you dont call this? You get an empty dockerfile

Reply:
Yes. What wouldyou expect to happen?

Reply:
Not sure, we need a better failure mode.

cc @mitchdenny for ideas.

Reply:
I half thought that a `ViteApp` would default to NpmPackageManager. And then if you wanted to replace it, you would call `WithYarnPackageManager`, which would replace the annotation.

But the issue with that approach is that `WithNpmPackageManager` adds the installer resource as well. And so if you also called `WithYarnPackageManager` now you'd have 2 "installer resources" added.

I guess we could just add the "npm" annotation by default and not the installer... but that doesn't feel right either.